Math/NumberTheory/Primes/Factorisation/Montgomery.hs

```diff
@@ -152,75 +155,88 @@ stdGenFactorisation primeBound sg digits n
 --
 --   @'curveFactorisation'@ requires that small prime factors have been stripped before. Also, it is
 --   unlikely to succeed if @n@ has more than one (really) large prime factor.
-curveFactorisation :: Maybe Integer                 -- ^ Lower bound for composite divisors
-                   -> (Integer -> Bool)             -- ^ A primality test
-                   -> (Integer -> g -> (Integer,g)) -- ^ A PRNG
-                   -> g                             -- ^ Initial PRNG state
-                   -> Maybe Int                     -- ^ Estimated number of digits of the smallest prime factor
-                   -> Integer                       -- ^ The number to factorise
-                   -> [(Integer,Int)]               -- ^ List of prime factors and exponents
+curveFactorisation
+  :: forall g.
```

Comment:
i like that this is parameterized over the RNG alg :)

Reply:
to clarify the comment on this function: the integer input to curveFactorization assumes "small" factors are absent from the factorization of the input, right?

Reply:
I have expanded the comment.

Reply:
ok, very nice

what RNG is default STD gen? :)

Reply:
`System.Random.mkStdGen` with an xored argument of `factorise` as a seed.